### DISCUSSION TITLE
Where to configure Cost / kwh

### DISCUSSION THREAD

**Discussion by luisfocosta** [David00/rpi-power-monitor#49] [Q&A] [answered]
Where can I adjust the cost per kwh which calculates the total cost in the Dashboard?
Thanks

**Comment by David00** [comment]
When you view the dashboard in Grafana, there is a small settings wheel near the top of the page. Click this settings wheel, then go to Variables.

You'll see the `cost_per_kWh` variable listed there, which you can edit as needed.

![image](https://user-images.githubusercontent.com/17997328/167887326-3c47f146-1a12-4367-a7ae-398de738f987.png)

**Reply by luisfocosta** [reply]
Awesome, thanks!
I assume only one value/rate  is possible. I will try to calculate cost using variable rates, maintained in a table.

**Comment by David00** [comment, marked as answer]
Yes, right now, only one rate is supported. I have thought about adding variable rates before, and the only way I could think of was to create a new measurement and store the cost of each unit of power as part of each calculation cycle.  I'm sure it's possible in Grafana and Influx, but even adding queries together is convoluted IMO.

**Reply by luisfocosta** [reply]
I'll give it a try nonetheless. I'd like to get a graph cost/hour which should be very interesting. Also considering solar.
For people with solar - like me - having a graph which shows the house potential dollar savings (by offsetting consumption with solar) would be great.
But, yeah, it's becoming a little complicated.

**Reply by David00** [reply]
That would be very cool! Even for me with grid tied solar, the local utility provides me with credits for excess energy that I export, but the price per kWh that they credit me is less than what I would pay for power that I import from them (like at night).

So, even though I don't have a variable rate schedule or time of use billings, I do have variable costs depending on which direction the power is flowing through the meter.

I am working on a complete overhaul and adding a user interface in a new version of my project, and I intend on supporting time of use/variable cost tracking in this new version. It's not finished yet though, and I haven't started on implementing the variable rate at all. But, since you ordered through my site, you'll be on the mailing list for the announcements when I'm ready to send them out.

**Reply by luisfocosta** [reply]
That's great! Keep me posted

**Reply by Splodgeink** [reply]
It would be really useful to add a fixed cost per day to this value over any given period, as we in the UK have a, now, considerable standing charge!
